pika/connection.py

```diff
@@ -370,13 +370,15 @@ def heartbeat(self, value):
         """
         :param value: desired connection heartbeat timeout for negotiation or
             None to accept broker's value. 0 turns heartbeat off.
+            Can be a callable that accepts server value and returns a new value
```

Comment:
Please add a test for passing callable heartbeat using `Parameters` class to the appropriate section in `test/unit/connection_parameters_tests.py`